src/coreComponents/physicsSolvers/wavePropagation/pygeosx/mesh.py

```diff
@@ -0,0 +1,297 @@
+    """Get mesh informations from GEOSX mesh for the seismic acquisition
+    
+    Parameters
+    ----------
+    problem :
+        A pygeosx.initialize object
+    
+    Return
+    ------
+    mesh :
+        A Mesh object
+    """
+    
+    node_manager = problem.get_group("domain/MeshBodies/mesh/Level0/nodeManager")
+
+    number_of_node = len(node_manager.get_wrapper("ReferencePosition").value().to_numpy())
+    
+    boundary_node_list = []    
+    for i in range(number_of_node):
+    	if node_manager.get_wrapper("domainBoundaryIndicator").value().to_numpy()[i]==1:
+    	     boundary_node_list.append(i)
+    	     
+
+    elem_manager = problem.get_group("domain/MeshBodies/mesh/Level0/ElementRegions/elementRegionsGroup/Region/elementSubRegions/cb")
+  
+    elem_speed  = problem.get_wrapper("FieldSpecifications/cellVelocity/scale").value()[0]
+    
+    number_of_elem = len(elem_manager.get_wrapper("nodeList").value().to_numpy())
+    
+    nb_node_per_elem = elem_manager.get_wrapper("numNodesPerElement").value()[0]
+    if nb_node_per_elem==8:
+    	disc_ord=1
+    elif nb_node_per_elem==64:
+    	disc_ord=3
+    elif nb_node_per_elem==216:
+    	disc_ord=5
+    
+    
+    Node_list = [Node(node_manager.get_wrapper("ReferencePosition").value().to_numpy()[i], i) for i in range(number_of_node)]
+    node_array=np.array(Node_list)
+    Elem_list = [Element(node_array[elem_manager.get_wrapper("nodeList").value().to_numpy()[i]].tolist(), 
+                         elem_speed, 
+                         elem_manager.get_wrapper("elementCenter").value().to_numpy()[i], 
+                         elem_manager.get_wrapper("elementVolume").value().to_numpy()[i], 
+                         i) 
+                         for i in range(number_of_elem)]
+   
+    mesh = Mesh(Elem_list, 
+                Node_list, 
+                node_array[boundary_node_list].tolist(), 
+                disc_ord)
+    
+    return mesh
+    
+    
+      
+    
+class Node:
+    """A class representing a node of the mesh
+    
+    Attributes
+    ----------
+    coords :
+        Node coordinate
+    
+    n :
+        The global number of the node
+    """
+        
+    def __init__(self, nodecoord, nodenumber):
+        """Constructor of Node
+        
+        Parameters
+        ----------
+        nodecoord :
+            Coordinates of the node
+        
+        nodenumber :
+            Global number of the node in the mesh
+        """
+        
+        self.x = nodecoord[0]                          
+        self.y = nodecoord[1]                          
+        self.z = nodecoord[2]                          
+        self.coords = np.array([self.x, self.y, self.z])         
+        self.n = nodenumber                            
+        
+    def getCoords(self):
+        return self.coords
+    
+    def x(self):
+    	return self.x
+    
+    def y(self):
+    	return self.y
+    
+    def z(self):
+    	return self.z
```

Comment:
This isn't a very important comment, but why duplicate the coordinates? I would just keep `self.coords`. Then have `x()` return `self.coords[ 0 ]`.

Reply:
Yes I was planning to remove all of these duplicates, I will do it today